### PR TITLE
boards: arm: Add steval_fcu001v1 board support

### PR DESCRIPTION
**Pull request by Navin-Sankar**
Add the BSP for STEVAL_FCU001V1 board with documentation.

Signed-off-by: NavinSankar Velliangiri <navin@linumiz.com>

### CODE REVIEW

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for the documentation.  Here are some recommended changes.

**Comment by Navin-Sankar** [comment]
Some checks haven't started yet.  @ioannisg

File: boards/arm/steval_fcu001v1/steval_fcu001v1.dts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f401Xe.dtsi>
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/steval_fcu001v1.dts]: Board is using STM32F401CC (with 64Kb RAM, 256K Flash), you need to add the matching stm32f401Xc.dtsi file (stm32f401Xe.dtsi declares 96Kb RAM, 512K Flash).

File: boards/arm/steval_fcu001v1/support/openocd.cfg [hunk 2]
```diff
@@ -0,0 +1,13 @@
+source [find board/st_nucleo_f4.cfg]
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/support/openocd.cfg]: I'm not familiar with external ST Link usage, is this really valid?

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by Navin-Sankar** [comment]
@erwango PR is closed by my mistake. Please re-open it

**Comment by erwango** [comment]
@Navin-Sankar can't do, sorry. Maybe @ioannisg / @galak ?

**Comment by Navin-Sankar** [comment]
@ioannisg @galak PR is closed by my mistake. Please re-open it.

**Comment by Navin-Sankar** [comment]
@ioannisg @galak Please re-open the PR.

**Comment by erwango** [comment]
@Navin-Sankar, for some reason we're not able to reopen this PR. Can you check if you have the ability to do it ? Otherwise please raise a new PR an we resume works where we stopped.

**Comment by ioannisg** [comment]
Is the branch deleted?

**Pull request closed without merging**